module/documents/active-effect.mjs

```diff
@@ -142,6 +143,20 @@ export default class ActiveEffect5e extends ActiveEffect {
       return current;
     }
 
+    // If attempting to apply active effect to empty MappingField entry, create it
+    if ( (current === undefined) && change.key.startsWith("system.") ) {
+      let keyPath = change.key;
+      let mappingField = field;
+      while ( !(mappingField instanceof MappingField) && mappingField ) {
+        if ( mappingField.name ) keyPath = keyPath.substring(0, keyPath.length - mappingField.name.length - 1);
+        mappingField = mappingField.parent;
+      }
+      if ( mappingField && (foundry.utils.getProperty(model, keyPath) === undefined) ) {
```

Comment:
I guess you must have tested it, but `mappingField.fieldPath` is not working correctly?

Reply:
That gets the path to the `MappingField` (e.g. `system.tools`), but that will already exist. We need the path to the specific entry within the field (e.g. `system.tools.thief`), which isn't available from one of the `DataFields` because it isn't static.